a10_octavia/controller/worker/flows/a10_listener_flows.py

```diff
@@ -143,41 +200,54 @@ def get_update_listener_flow(self):
         :returns: The flow for updating a listener
         """
         update_listener_flow = linear_flow.Flow(constants.UPDATE_LISTENER_FLOW)
-        update_listener_flow.add(lifecycle_tasks.ListenersToErrorOnRevertTask(
-            requires=[constants.LOADBALANCER, constants.LISTENERS]))
+        update_listener_flow.add(lifecycle_tasks.ListenerToErrorOnRevertTask(
+            requires=constants.LISTENER))
         update_listener_flow.add(a10_database_tasks.GetVThunderByLoadBalancer(
             requires=constants.LOADBALANCER,
             provides=a10constants.VTHUNDER))
-        update_listener_flow.add(virtual_port_tasks.ListenersUpdate(
-            requires=[constants.LOADBALANCER, constants.LISTENERS, a10constants.VTHUNDER]))
-
+        update_listener_flow.add(*self.get_listener_update_cert_template_subflow())
+        update_listener_flow.add(virtual_port_tasks.ListenerUpdate(
+            requires=[constants.LOADBALANCER, constants.LISTENER, a10constants.VTHUNDER]))
         update_listener_flow.add(database_tasks.UpdateListenerInDB(
             requires=[constants.LISTENER, constants.UPDATE_DICT]))
-        update_listener_flow.add(database_tasks.
-                                 MarkLBAndListenersActiveInDB(
-                                     requires=[constants.LOADBALANCER,
-                                               constants.LISTENERS]))
+        update_listener_flow.add(a10_database_tasks.MarkLBAndListenerActiveInDB(
+                                 requires=[constants.LOADBALANCER,
+                                           constants.LISTENER]))
 
         return update_listener_flow
 
+    def _get_cert_update_subflow(self):
+        get_cert_update_subflow = linear_flow.Flow(a10constants.UPDATE_CERT_TEMPLATE_SUBFLOW)
+        get_cert_update_subflow.add(cert_tasks.SSLKeyUpdate(
+            requires=(a10constants.CERT_DATA, a10constants.VTHUNDER)))
+        get_cert_update_subflow.add(cert_tasks.ClientSSLTemplateUpdate(
+            requires=(a10constants.CERT_DATA, a10constants.VTHUNDER)))
+        return get_cert_update_subflow
+
+    def get_listener_update_cert_template_subflow(self):
+        get_listener_update_cert_subflow = graph_flow.Flow(
+            a10constants.UPDATE_LISTENER_CERT_SUBFLOW)
+        cert_update_task = cert_tasks.SSLCertUpdate(
+            requires=[constants.LOADBALANCER, constants.LISTENER, a10constants.VTHUNDER],
+            provides=a10constants.CERT_DATA)
+        cert_update_template_task = self._get_cert_update_subflow()
+        get_listener_update_cert_subflow.add(cert_update_task, cert_update_template_task)
+        get_listener_update_cert_subflow.link(
+            cert_update_task, cert_update_template_task, decider=self._is_terminated_https)
```

Comment:
This graph_flow and decider should be in get_update_listener_flow. Similar to earlier comments.

Reply:
Please refer my above comments